src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,119 @@
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private final Importance DEFAULT_IMPORTANCE = Importance.LITTLE_IMPORTANT;
+
+    private final TaskRepository taskRepository;
+    private final UserRepository userRepository;
+
+    public TaskService(TaskRepository taskRepository, UserRepository userRepository) {
+        this.taskRepository = taskRepository;
+        this.userRepository = userRepository;
+    }
+
+    public List<Task> getTasks() {
+        archiveTasks();
+        return taskRepository.findAll();
+    }
+
+    public List<Task> getAllLoggedUserTasks(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUser(loggedUser);
+    }
+
+    public List<Task> getAllLoggedUserTasksOrderByImportance(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserOrderByImportance(loggedUser);
+    }
+
+    public List<Task> getLoggedUserParticularTasks(Principal principal, Importance importance) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserIdaAndImportance(loggedUser.getId(), importance);
+    }
+
+    public List<Task> getUserTasks(Long userId) {
+        archiveTasks();
+        return taskRepository.findALLByUserId(userId);
+    }
+
+    public Task getTaskById(Long taskId) {
+        archiveTasks();
+        Optional<Task> task = taskRepository.findById(taskId);
+        return task.get();
+    }
+
+    public Task addTask(Task task, Principal principal) {
+        task.setImportance(DEFAULT_IMPORTANCE);
+        task.setCreationDate(LocalDateTime.now());
+        task.setExpirationDate(endOfTheDay());
+        task.setUser(userRepository.findByLogin(principal.getName()).get());
+        return taskRepository.save(task);
+    }
+
+    public Task updateTask(Task task, Long id) {
+        Task updatedTask = taskRepository.findById(id).get();
+        updatedTask.setImportance(task.getImportance());
+        updatedTask.setDescription(task.getDescription());
+        updatedTask.setArchived(task.isArchived());
+        updatedTask.setDone(task.isDone());
+        return taskRepository.save(updatedTask);
+    }
+
+    public void deleteTask(Long id) {
+        taskRepository.deleteById(id);
+    }
+
+
+    public List<Task> getTaskByImportance(Importance importance) {
+        return taskRepository.findAllByImportance(importance);
+    }
+
+
+    public LocalDateTime endOfTheDay() {
```

Comment:
This doesnt allow to plan tasks for future.

Reply:
It's not for plan task for the future but for automatically archive task at the end of the day.

Reply:
Yeah, I know, but this code doesn't work for tasks planned for future. I mean, every task will be archived by the end of the day it was created on.